File: 1349-check-if-it-is-a-straight-line/1349-check-if-it-is-a-straight-line.java
class Solution {
    public boolean checkStraightLine(int[][] coordinates) {

        int x1 = coordinates[0][0] , y1 = coordinates[0][1];
        int x2 = coordinates[1][0] , y2 = coordinates[1][1];
        
        for(int i= 2 ; i < coordinates.length ; i++){
            int x0 = coordinates[i][0]; 
            int y0 = coordinates[i][1];

            if ((x1 - x0) * (y2 - y0) != (y1 - y0) * (x2 - x0))
{
                return false;
            }
        }
        return true;
    }
}

File: 1349-check-if-it-is-a-straight-line/1349-check-if-it-is-a-straight-line.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:

        lst = coordinates

        x_side = lst[0][0] - lst[1][0]
        y_side = lst[0][1] - lst[1][1]
        
        if(y_side):
            inc = x_side/y_side
        else:
            inc = float('inf')
              
        for i in range(len(lst)-1):
            x1,y1 = lst[i]
            x2,y2 = lst[i+1]

            dx = x1-x2 
            dy = y1-y2

            # 분모가 0일 때
            if (y1-y2) == 0:
                if inc != float('inf'):
                    return False
                continue

            if ((x1-x2) / (y1-y2)) != inc:
                return False

        return True

        